src/chocolatey/infrastructure/services/XmlService.cs

```diff
@@ -43,14 +44,46 @@ public XmlType deserialize<XmlType>(string xmlFilePath)
                 () =>
                 {
                     var xmlSerializer = new XmlSerializer(typeof(XmlType));
-                    var xmlReader = XmlReader.Create(new StringReader(_fileSystem.read_file(xmlFilePath)));
-                    if (!xmlSerializer.CanDeserialize(xmlReader))
+                    using (var fileStream = _fileSystem.open_file_readonly(xmlFilePath))
+                    using (var fileReader = new StreamReader(fileStream))
+                    using (var xmlReader = XmlReader.Create(fileReader))
                     {
-                        this.Log().Warn("Cannot deserialize response of type {0}", typeof(XmlType));
-                        return default(XmlType);
-                    }
+                        if (!xmlSerializer.CanDeserialize(xmlReader))
+                        {
+                            this.Log().Warn("Cannot deserialize response of type {0}", typeof(XmlType));
+                            return default(XmlType);
+                        }
+
+                        try 
+                        {
+                            return (XmlType)xmlSerializer.Deserialize(xmlReader);
+                        } 
+                        catch(InvalidOperationException ex)
+                        {
+                            // Check if its just a malformed document.
+                            if (ex.Message.Contains("There is an error in XML document"))
+                            {
+                                // If so, check for a backup file and try an parse that.
+                                if (_fileSystem.file_exists(xmlFilePath + ".backup"))
+                                {
+                                    using (var backupStream = _fileSystem.open_file_readonly(xmlFilePath + ".backup"))
+                                    using (var backupReader = new StreamReader(backupStream))
+                                    using (var backupXmlReader = XmlReader.Create(backupReader))
+                                    {
+                                        var validConfig = (XmlType)xmlSerializer.Deserialize(backupXmlReader);
 
-                    return (XmlType)xmlSerializer.Deserialize(xmlReader);
+                                        // If there's no errors and it's valid, go ahead and replace the bad file with the backup.
+                                        if(validConfig != null)
+                                        {
+                                            _fileSystem.copy_file(xmlFilePath + ".backup", xmlFilePath, overwriteExisting: true);
+                                        }
+                                        return validConfig;
+                                    }
+                                }
+                            } 
+                            throw;
+                        }
+                    }

```

Comment:
So this is the meat - just compares the stream versus the file before it creates the old file? This would be great in low access environments so there is no error message when it can't create the file.